scripts/managed-services/get-kafka-instance-state/get-kafka-instance-state.sh

```diff
@@ -0,0 +1,23 @@
+#!/bin/bash
+
+function main() {
+  local kafkaId=$1
+  local managedKafkaCR=$(oc -n kafka-$kafkaId get managedkafka -o name)
+  local kafkaCR=$(oc -n kafka-$kafkaId get kafka -o name)
+  local kafkaStatefulSet=$(oc -n kafka-$kafkaId get statefulset -l app.kubernetes.io/name=kafka -o name)
+  local timestamp=$(date +%s)
+  local inspectDir="inspect.kafka-$kafkaId.$timestamp"
+  local namespaceCheck=$(oc get namespace kafka-$kafkaId)
+
+  oc -n kafka-$kafkaId adm inspect ns/kafka-$kafkaId $managedKafkaCR $kafkaCR --dest-dir $inspectDir > /dev/null 2>&1
+
+  oc exec -n kafka-$kafkaId $kafkaStatefulSet -c kafka -- sh /opt/kafka/bin/kafka-topics.sh --bootstrap-server localhost:9096 --list > ${inspectDir}/kafka-topics.txt 2>&1
+
+  oc exec -n kafka-$kafkaId $kafkaStatefulSet -c kafka -- sh /opt/kafka/bin/kafka-consumer-groups.sh --bootstrap-server localhost:9096 --all-groups --describe  > ${inspectDir}/kafka-consumer-groups.txt 2>&1
+
+  oc exec -n kafka-$kafkaId $kafkaStatefulSet -c kafka -- sh /opt/kafka/bin/kafka-acls.sh --bootstrap-server localhost:9096 --list > ${inspectDir}/kafka-acls.txt 2>&1
+
```

Comment:
Possibly out of scope for this script but I think it would be useful to capture the k8s setup when the commands were run.

```
oc get pods -n $kafkaId -o json > ${inspectDir}/namespace.json
```

Reply:
Or is that thought just a duplicate of the `oc adm inspect` ?

Reply:
`ocm adm inspect` captures both logs and the state of the pod resource (as yaml).

```
ind inspect.local.1420828432188943909/namespaces/kafka-c52ajhf2ek06ri44ife0/pods/mykafka-dev-zookeeper-0/
inspect.local.1420828432188943909/namespaces/kafka-c52ajhf2ek06ri44ife0/pods/mykafka-dev-zookeeper-0/
inspect.local.1420828432188943909/namespaces/kafka-c52ajhf2ek06ri44ife0/pods/mykafka-dev-zookeeper-0//mykafka-dev-zookeeper-0.yaml
inspect.local.1420828432188943909/namespaces/kafka-c52ajhf2ek06ri44ife0/pods/mykafka-dev-zookeeper-0//zookeeper
inspect.local.1420828432188943909/namespaces/kafka-c52ajhf2ek06ri44ife0/pods/mykafka-dev-zookeeper-0//zookeeper/zookeeper
inspect.local.1420828432188943909/namespaces/kafka-c52ajhf2ek06ri44ife0/pods/mykafka-dev-zookeeper-0//zookeeper/zookeeper/logs
inspect.local.1420828432188943909/namespaces/kafka-c52ajhf2ek06ri44ife0/pods/mykafka-dev-zookeeper-0//zookeeper/zookeeper/logs/current.log
inspect.local.1420828432188943909/namespaces/kafka-c52ajhf2ek06ri44ife0/pods/mykafka-dev-zookeeper-0//zookeeper/zookeeper/logs/previous.log
inspect.local.1420828432188943909/namespaces/kafka-c52ajhf2ek06ri44ife0/pods/mykafka-dev-zookeeper-0//zookeeper/zookeeper/logs/previous.insecure.log```

Reply:
Ignore me then 😁